### PR TITLE
LSR-1005 Android: PortletDisplayScreenlet

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

File: android/library/src/main/assets/gallery.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+function addClickToCards() {
+	var cards = document.getElementsByClassName('card');
+    var allImgSrc = "";
```
Comment (nhpatt) [inline android/library/src/main/assets/gallery.js]: please, ⌥⌘L

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+	public enum WebType {LIFERAY_AUTHENTICATED, LIFERAY, CUSTOM}
+
+	public static class Builder {
+
+		private String portletUrl;
+		private List<String> localJs;
+		private List<String> localCss;
+		private List<String> remoteJs;
+		private List<String> remoteCss;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java]: CSS and JS

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+		private boolean isThemeEnabled;
+		private WebType webType;
+		private CordovaLifeCycleObserver observer;
+		private boolean isCordovaEnabled;
+
+		public Builder(String portletUrl) {
+			super();
+
+			this.portletUrl = portletUrl;
+			localJs = new ArrayList<>();
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java]: why not directly initialized?

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+			localCss = new ArrayList<>();
+			remoteJs = new ArrayList<>();
+			remoteCss = new ArrayList<>();
+			localRawJs = new ArrayList<>();
+			localRawCss = new ArrayList<>();
+			this.isThemeEnabled = true;
+			this.isCordovaEnabled = false;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java]: this is the default

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+		public Builder(String portletUrl, List<String> localJs, List<String> localCss,
+			List<String> remoteJs, List<String> remoteCss, boolean isThemeEnabled) {
+			this.portletUrl = portletUrl;
+			this.localJs = localJs;
+			this.localCss = localCss;
+			this.remoteJs = remoteJs;
+			this.remoteCss = remoteCss;
+			this.isThemeEnabled = isThemeEnabled;
+			this.isCordovaEnabled = false;
+			this.webType = WebType.LIFERAY_AUTHENTICATED;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletConfiguration.java]: why not directly?

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletDisplayScreenlet.java [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+			String finalUrl = buildPortletUrl(portletConfiguration.getPortletUrl());
+			String body = buildBody();
+
+			String screensJs = new AssetReader(getContext()).read(R.raw.screens);
+
+			getViewModel().addScript(new JsScript("Screens.js", screensJs));
+
+			if (portletConfiguration.isThemeEnabled() && !portletConfiguration.getWebType()
+				.equals(PortletConfiguration.WebType.CUSTOM)) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/PortletDisplayScreenlet.java]: invert the comparison: PortletConfiguration.WebType.CUSTOM.equals... (can't be null)

File: android/library/src/main/java/com/liferay/mobile/screens/portlet/util/JsScript.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+ * @author Sarai Díaz García
+ */
+public class JsScript implements InjectableScript {
+
+	private String name;
+	private String content;
+
+	public JsScript(String name, String js) {
+		this.name = name;
+		content = js;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/util/JsScript.java]: why not this.js = js

Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/portlet/util/JsScript.java]: or this.content = content

File: android/library/src/main/java/com/liferay/mobile/screens/util/AssetReader.java [hunk 8]
```diff
@@ -25,24 +25,44 @@
...
+		this.context = context;
+	}
 
-			reader.close();
+	public String read(int fileId) {
+		try {
+			InputStream in = context.getResources().openRawResource(fileId);
+			return getFileContent(in);
+		} catch (Exception e) {
+			e.printStackTrace();
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/util/AssetReader.java]: log or throw

File: android/library/src/main/java/com/liferay/mobile/screens/util/AssetReader.java [hunk 9]
```diff
@@ -25,24 +25,44 @@
...
-			return null;
+			return "";
 		}
 	}
-}
+
+	private String getFileContent(InputStream inputStream) throws Exception {
+		BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+		StringBuilder sb = new StringBuilder();
+		String mLine = reader.readLine();
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/util/AssetReader.java]: tsk, tsk

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/imagegallery/DetailImageActivity.java [hunk 10]
```diff
@@ -5,30 +5,69 @@
...
+	@Override
+	public void onSwipeLeft() {
+		Picasso.with(getApplicationContext()).load(allImgSrc[imgSrcPosition + 1]).into(detailedImageView);
+		imgSrcPosition++;
+
+	}
+
+	@Override
+	public void onSwipeRight() {
+		Picasso.with(getApplicationContext()).load(allImgSrc[imgSrcPosition - 1]).into(detailedImageView);
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/imagegallery/DetailImageActivity.java]: PicassoFactory?

File: android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/portlet/cordova/ScreensCordovaWebView.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+		CordovaWebView cordovaWebView = cordovaActivity.initCordova();
+
+		observer.setListener(this);
+
+		webView = ((WebView) cordovaWebView.getView());
+	}
+
+	public void onEvent(CordovaEvent event) {
+		if (listener != null) {
+			if (event.getEventType().equals(CordovaEvent.CordovaEventType.PAGE_STARTED)) {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/portlet/cordova/ScreensCordovaWebView.java]: invert and extract variable

File: android/library/src/main/java/org/apache/cordova/ScreensCordovaActivity.java [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+	}
+
+	public void onSaveInstanceState(Bundle outState) {
+		this.cordovaInterface.onSaveInstanceState(outState);
+	}
+
+	public void onConfigurationChanged(Configuration newConfig) {
+		super.onConfigurationChanged(newConfig);
+		if(this.appView != null) {
```
Comment (nhpatt) [inline android/library/src/main/java/org/apache/cordova/ScreensCordovaActivity.java]: formatting

File: android/library/src/main/java/org/apache/cordova/ScreensCordovaActivity.java [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+
+		return true;
+	}
+
+	public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+		try {
+			this.cordovaInterface.onRequestPermissionResult(requestCode, permissions, grantResults);
+		} catch (JSONException ex) {
+			LOG.d(TAG, "JSONException: Parameters fed into the method are not valid");
+			ex.printStackTrace();
```
Comment (nhpatt) [inline android/library/src/main/java/org/apache/cordova/ScreensCordovaActivity.java]: :P

**Comment by victorg1991** [comment]
Applied changes :D

**Comment by nhpatt** [comment]
👍

**Pull request merged**